controllers/machinehealthcheck_controller.go

```diff
@@ -269,7 +316,31 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 			t.string(),
 		)
 	}
+
 	for _, t := range healthy {
+
+		if m.Spec.RemediationTemplate != nil {
+
+			remediationObj := &corev1.ObjectReference{
+				APIVersion: m.Spec.RemediationTemplate.APIVersion,
+				Kind:       strings.TrimSuffix(m.Spec.RemediationTemplate.Kind, external.TemplateSuffix),
+				Name:       t.Machine.Name,
+			}
+
+			// Get remediation request object
+			obj, err := external.Get(ctx, r.Client, remediationObj, t.Machine.Namespace)
+			if err != nil && !apierrors.IsNotFound(errors.Cause(err)) {
+				return ctrl.Result{}, errors.Wrapf(err, "Failed to fetch remediation request")
+			}
```

Comment:
Can we handle the IsNotFound separately by just calling `continue`? This way we don't have to check if obj != nil, but only its deletion timestamp